python_ta/transforms/type_inference_visitor.py

```diff
@@ -258,17 +258,39 @@ def register_type_constraints_setter():
     return type_visitor
 
 
+def _populate_local_env(node):
+    """Helper to populate locals attributes in type environment of given node."""
+    for var_name in node.locals:
+        try:
+            var_value = node.type_environment.lookup_in_env(var_name)
+        except KeyError:
+            var_value = TYPE_CONSTRAINTS.fresh_tvar()
+        node.type_environment.locals[var_name] = var_value
+
+
 def _set_environment(node):
     node.type_environment = Environment(locals_={name: TYPE_CONSTRAINTS.fresh_tvar() for name in node.locals},
                                         globals_={name: TYPE_CONSTRAINTS.fresh_tvar() for name in node.globals})
     if isinstance(node, astroid.FunctionDef):
         node.type_environment.locals['return'] = TYPE_CONSTRAINTS.fresh_tvar()
 
 
+def _set_module_environment(node):
+    """Method to set environment of a Module node."""
+    node.type_environment = Environment(globals_={name: TYPE_CONSTRAINTS.fresh_tvar() for name in node.globals})
+    _populate_local_env(node)
+
+
+def _set_funcion_def_environment(node):
+    """Method to set environment of a FunctionDef node."""
+    _populate_local_env(node)
+    node.type_environment.locals['return'] = TYPE_CONSTRAINTS.fresh_tvar()
```

Comment:
This is why I used to have the isinstance check, but I feel like this method will never be called on a non-FunctionDef node by nature of the way these methods are registered in lines 294-295.
` visitor.register_transform(astroid.FunctionDef, _set_funcion_def_environment)
 visitor.register_transform(astroid.Module, _set_module_environment)`